### PR TITLE
Setup hub for working with database

### PR DESCRIPTION
**Pull request by lukemelia**
- Migrations support
- DI-aware DatabaseManager to access PG client
- Test database setup (with transactional rollbacks between tests)

### CODE REVIEW

File: packages/hub/config/structure.sql [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+--
+
+ALTER TABLE ONLY public.prepaid_card_customizations
+    ADD CONSTRAINT prepaid_card_customizations_pattern_id_fkey FOREIGN KEY (pattern_id) REFERENCES public.prepaid_card_header_patterns(id);
+
+
+--
+-- PostgreSQL database dump complete
+--
+
```
Comment (backspace) [inline packages/hub/config/structure.sql:162]: Process note: I thought it made sense to just ignore this file, as it’s auto-generated 🤓

**Review by backspace — APPROVED** [review]
I made a couple of minor suggestions, but nothing blocking. This seems like a sensible implementation of test database setup/teardown that fits within the existing structure. Caveat that while I’d normally like to run this kind of thing locally at least once, I haven’t been able to because of mysterious failures… probably related to recently updating to Big Sur 😭

File: packages/hub/migrations/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Migrations currently must be authored in Javascript. `node-pg-migrate` has typescript support but I was unable to get it to work with our repository.
```
[the remark this answers is not in the document]
Comment (lukemelia) [reply packages/hub/migrations/README.md:1]: Hopefully someone with more time and better understanding can fix this!

File: packages/hub/bin/setup-test-db.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const dbConfig = config.get('db');
+  try {
+    console.log('Dropping hub_test db...');
+    let result = await exec(`dropdb hub_test`);
+    console.log(result);
+  } catch (e) {
+    // ok if this fails
+  }
+  console.log('Creating new hub_test db...');
+  let result = await exec(`createdb hub_test`);
```
[the remark this answers is not in the document]
Comment (lukemelia) [reply packages/hub/bin/setup-test-db.ts:25]: Good idea.

File: packages/hub/config/default.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+
+  "db": {
+
+      "url": "postgres://postgres:postgres@localhost:5432/hub_development",
+      "migration-filename-format": "utc",
+      "ignore-pattern": "README.md|.*\\.ts|.*\\.js\\.map"
+  }
```
[the remark this answers is not in the document]
Comment (lukemelia) [reply packages/hub/config/default.json:8]: I blame my editor. :) Will fix.

**Pull request merged**